Makefile

```diff
@@ -4,37 +4,51 @@ PARAMS_FILE   := cloudformation/parameters.json
 EXCLUDE_DIRS  := .git|tests|cloudformation|requirements|$(STAGING_DIR)|$(BUILDS_DIR)
 PIP_COMMAND   := pip install -r
 
-.PHONY: init create-stack update-stack delete-stack describe-stack invoke test clean build _check-arn deploy
+.PHONY: init create-stack update-stack delete-stack describe-stack invoke test clean build update _check_config _check_stack_exists _is_user_authenticated
 
 # Read the cloudformation/parameters.json file for the ProjectName and EnvionmentName.
 # Use these to name the CloudFormation stack.
-PROJECT_NAME = $(shell cat $(PARAMS_FILE) | python -c 'import sys, json; j = [i for i in json.load(sys.stdin) if i["ParameterKey"]=="ProjectName"][0]["ParameterValue"]; print(j)')
-ENVIRONMENT_NAME = $(shell cat $(PARAMS_FILE) | python -c 'import sys, json; j = [i for i in json.load(sys.stdin) if i["ParameterKey"]=="EnvironmentName"][0]["ParameterValue"]; print(j)')
-STACK_NAME = $(PROJECT_NAME)-$(ENVIRONMENT_NAME)-stack
+PROJECT_NAME := $(shell cat $(PARAMS_FILE) | jq -r '.[] | select(.ParameterKey == "ProjectName") | .ParameterValue')
+ENVIRONMENT_NAME := $(shell cat $(PARAMS_FILE) | jq -r '.[] | select(.ParameterKey == "EnvironmentName") | .ParameterValue')
+S3_KEY_NAME := $(shell cat $(PARAMS_FILE) | jq -r '.[] | select(.ParameterKey == "S3DeploymentFileKey") | .ParameterValue')
```

Comment:
I think instead of this it makes sense to create a code bucket for the account. And the key name can be a combination of project/env.

Reply:
I think it seems like a bit of a waste to create an entire bucket just for one Lambda function to be deployed through, even though the user could override it. Perhaps the default could be something like `lambda-deployments-${AWS::AccountId}` or similar, so that all deployments to the same account are the same. This gets a little bit tricky though in implementation. 

I guess the CLI script could check if S3_BUCKET_NAME is empty, and if so it will create a new bucket through `aws s3 create-bucket`. Then we need similar logic in the CFN template? Or do we have the makefile update the `parameters.json` with the new bucket?

Reply:
I would like to move to a standard `lambda-deployments-${AWS::AccountId}` thing eventually, but let's leave as is for now. I don't know that it's worth bothering with the override even... it feels a bit YAGNI to me.